e2e/test/iothub/service/DevicesClientE2ETests.cs

```diff
@@ -63,7 +63,7 @@ public async Task DevicesClient_AddAndRemoveDeviceWithScope()
             string edgeId2 = _idPrefix + Guid.NewGuid();
             string deviceId = _idPrefix + Guid.NewGuid();
 
-            using var serviceClient = new IotHubServiceClient(TestConfiguration.IotHub.ConnectionString);
```

Comment:
By instantiating the service client less, we should be creating fewer HttpClient instances and [leaking](https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/http/httpclient-guidelines#recommended-use) fewer sockets.

I left any service client instantiations that used custom settings (like proxy).